ansible/roles/nginx/tasks/certbot.yml

```diff
@@ -6,23 +6,44 @@
     executable: pip3
 
 - name: Register certbot
+  run_once: true
   shell: |
     certbot -n register --agree-tos --email {{ letsencrypt_register_email }} --work-dir {{ nginx_letsencrypt_workdir }} --logs-dir {{ nginx_letsencrypt_logsdir }} --config-dir {{ nginx_letsencrypt_configdir }}
     touch {{ nginx_letsencrypt_workdir }}/.registered
   args:
     creates: "{{ nginx_letsencrypt_workdir }}/.registered"
 
-- name: Get certificate
+- name: "Get {{ hostname }} certificate info"
+  shell: openssl x509 -in "{{ nginx_letsencrypt_configdir }}/live/{{ hostname }}/cert.pem" -noout -dates | awk -F= '/notAfter=/ {print $2}'
+  register: cert_info
+
+- name: "Show {{ hostname }} certificate info"
+  debug:
+    msg: "{{ cert_info }}"
+
+- name: "Check {{ hostname }} certificate expire date in days"
+  set_fact:
+    cert_days_left: "{{ (( (cert_info.stdout | to_datetime('%b %d %H:%M:%S %Y %Z')) - (now('%Y-%m-%d %H:%M:%S')) )).days }}"
+  when: cert_info.stderr == ''
```

Comment:
Not sure why we need this to create the new certificate. Shouldn't the cron job do it? I mean, if there's no certificate on the machine, create a new one. Otherwise, leave the script to renew it.

Reply:
You are right